mods/walls/init.lua

```diff
@@ -25,6 +25,56 @@ walls.register = function(wall_name, wall_desc, wall_texture_table, wall_mat, wa
 		walkable = true,
 		groups = { cracky = 3, wall = 1, stone = 2 },
 		sounds = wall_sounds,
+				after_place_node = function(pos, placer, itemstack, pointed_thing)
+			local pos_under = {x = pos.x, y = pos.y - 1, z = pos.z}
+			local pos_above = {x = pos.x, y = pos.y + 1, z = pos.z}
+			local node_under = minetest.get_node(pos_under).name
+			if minetest.get_item_group(node_under, "wall") == 1 then
+					local node_under_can = string.gsub(node_under, "_full$", "")
+					minetest.set_node(pos_under, {name = node_under_can .. "_full"})
+			end
+
+			local node_above = minetest.get_node(pos_above).name
+			if minetest.get_item_group(node_above, "wall") == 1 then
+					minetest.set_node(pos, {name = wall_name .. "_full"})
+			end
+		end,
+		after_dig_node = function(pos, oldnode, oldmetadata, digger)
+			local pos_under = {x = pos.x, y = pos.y - 1, z = pos.z}
+			local node_under = string.gsub(minetest.get_node(pos_under).name, "_full$", "")
+			if minetest.get_item_group(node_under, "wall") == 1 and
+					digger and digger:is_player() then
+				minetest.set_node(pos_under, {name = node_under})
+			end
+		end,
+	})
+
+	minetest.register_node(wall_name .. "_full", {
+		description = wall_desc,
+		drawtype = "nodebox",
+		node_box = {
+			type = "connected",
+			fixed = {{-1/4, -1/2, -1/4, 1/4, 1/2, 1/4}},
```

Comment:
Excuse the nitpick :-)

No need for a table of nodeboxes with a single nodebox.
It's better to get rid of the extra `{}`s to not waste the memory (The definition is stored for the entire server lifetime).
This is only a little here, but can add up as people copy MTG code everywhere.

Same for the other nodebox definitions below of course.

Reply:
I will attend to this soon, thank you. Sorry so long in responding